src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -261,18 +261,33 @@ internal static List<string> GetModuleFilePaths(string rootScriptPath, string fu
         {
             List<string> modulePaths = new List<string>();
             string functionFolder = Path.Combine(rootScriptPath, functionName);
+            string rootModuleDirectory = Path.Combine(rootScriptPath, PowerShellConstants.ModulesFolderName);
             string moduleDirectory = Path.Combine(functionFolder, PowerShellConstants.ModulesFolderName);
-            if (Directory.Exists(moduleDirectory))
+            string[] searchDirectories = new string[] { rootModuleDirectory, moduleDirectory };
+            modulePaths.AddRange(AddToModulePaths(searchDirectories));
+            return modulePaths;
+        }
+
+        internal static List<string> AddToModulePaths(string[] directories)
+        {
+            List<string> paths = new List<string>();
+            for (int i = 0; i < directories.Length; i++)
             {
-                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
-                    PowerShellConstants.ModulesManifestFileExtensionPattern));
-                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
-                    PowerShellConstants.ModulesBinaryFileExtensionPattern));
-                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
-                    PowerShellConstants.ModulesScriptFileExtensionPattern));
+                string currentDirectory = directories[i];
+                if (Directory.Exists(currentDirectory))
+                {
+                    paths.AddRange(Directory.GetFiles(currentDirectory,
+                        PowerShellConstants.ModulesManifestFileExtensionPattern,
+                        SearchOption.AllDirectories));
```

Comment:
@tohling How would you want to handle "duplicates"? Usually I would say "closest to the function" wins. 
However, PowerShell can import multiple versions of the same module into one environment. It is then up to the user to decide which one he/she wants to use. That said, because Azure Functions right now imports everything automatically, this won't necessarily work. 
Hence https://github.com/Azure/azure-webjobs-sdk-script/issues/897